src/components/Card.tsx

```diff
@@ -11,51 +23,115 @@ export const Card = (item: History): JSX.Element => {
 
   const JuzCard = (): JSX.Element => {
     return (
-      <div className={cardClassnames.container}>
+      <>
         <p className={cardClassnames.title}>Juz {item.juz}</p>
-        <p className={cardClassnames.data}>Murojaah by using {item.murojaahMethod}</p>
-        <p className={cardClassnames.data}>Total Murojaah is {item.totalMurojaah}</p>
+        <p className={cardClassnames.data}>Murojaah {Show(item.approachId)}</p>
+        <p className={cardClassnames.data}>
+          Repeated {item.repeat} {repeatSuffix}
+        </p>
         <p className={cardClassnames.date}>{formatDatetime(item.occuredAt)}</p>
-      </div>
+      </>
     );
   };
 
   const SurahCard = (): JSX.Element => {
     return (
-      <div className={cardClassnames.container}>
+      <>
         <p className={cardClassnames.title}>
           Surah {item.surah} {item.surahName}
         </p>
-        <p className={cardClassnames.data}>Murojaah by using {item.murojaahMethod}</p>
-        <p className={cardClassnames.data}>Total Murojaah is {item.totalMurojaah}</p>
-        <p className={cardClassnames.date}>{item.occuredAt}</p>
-      </div>
+        <p className={cardClassnames.data}>Murojaah {Show(item.approachId)}</p>
+        <p className={cardClassnames.data}>
+          Repeated {item.repeat} {repeatSuffix}
+        </p>
+        <p className={cardClassnames.date}>{formatDatetime(item.occuredAt)}</p>
+      </>
     );
   };
 
   const AyahCard = (): JSX.Element => {
     return (
-      <div className={cardClassnames.container}>
+      <>
         <p className={cardClassnames.title}>
           Ayah {item.start} to {item.end}
         </p>
         <p className={cardClassnames.data}>
           Surah {item.surah} {item.surahName}
         </p>
-        <p className={cardClassnames.data}>Murojaah by using {item.murojaahMethod}</p>
-        <p className={cardClassnames.data}>Total Murojaah is {item.totalMurojaah}</p>
-        <p className={cardClassnames.date}>{item.occuredAt}</p>
+        <p className={cardClassnames.data}>Murojaah {Show(item.approachId)}</p>
+        <p className={cardClassnames.data}>
+          Repeated {item.repeat} {repeatSuffix}
+        </p>
+        <p className={cardClassnames.date}>{formatDatetime(item.occuredAt)}</p>
+      </>
+    );
+  };
+
+  const BaseCard = (children: JSX.Element): JSX.Element => {
```

Comment:
please take this out of Card component

Reply:
currently will mark it with some tag for tech debt